### PR TITLE
`send` - main page

### LINKED ISSUE #55: `send` - main page
#### What would you like to be added:
Ability to send token
#### Why is this needed:
- [ ] Token Selector
- [ ] Balance display with max button
- [ ] Address field
- [ ] Transaction fee

image

### PR DESCRIPTION
**Pull request by thedoublejay**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:
1. Balances page -> Click -> Links to Send page
2. Send Page
3. QR Code scanner
4. Form validation
   - Amount only accepts numbers with decimals
   - Both address and amount are required
5. `PrimaryButton` component that has ability to have disabled effect
6. `MAX` button that sets amount to token amount

Pending items:
1. Unit tests
2. E2E tests
3. Will merge submit logic once this is merged to main

#### Which issue(s) does this PR fixes?:

Fixes #55 and #208

#### Additional comments?:
<div>

</div>

### CODE REVIEW

File: screens/AppNavigator/screens/BalancesScreen/SendScreen/SendScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+        }} onCloseModal={() => setIsQRScanning(false)} isQRScanning={isQRScanning}
+      />
+      <AddressRow control={control} openQRScanner={() => setIsQRScanning(true)} />
+      <AmountRow
+        token={token} control={control} onMaxPress={async (amount) => {
+          setValue('amount', amount)
+          await trigger('amount')
+        }}
+      />
+      <PrimaryButton disabled={!isValid} title='Send' onPress={onSubmit}>
```
Comment (ivan-zynesis) [inline screens/AppNavigator/screens/BalancesScreen/SendScreen/SendScreen.tsx:61]: Not familiar with this `useForm` API, but I cant identify what's the logic that decide whether it `isValid` here

File: screens/AppNavigator/screens/BalancesScreen/SendScreen/SendScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    </ScrollView>
+  )
+}
+
+function AddressRow ({ control, openQRScanner }: AddressForm): JSX.Element {
+  return (
+    <>
+      <Text
+        style={tailwind('text-sm text-gray-300 font-bold pl-4 mt-4 mb-2')}
+      >{translate('screens/SendScreen', 'TO ADDRESS')}
```
Comment (ivan-zynesis) [inline screens/AppNavigator/screens/BalancesScreen/SendScreen/SendScreen.tsx:73]: I think this text should be in black color

File: screens/AppNavigator/screens/BalancesScreen/SendScreen/SendScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+
+type Props = StackScreenProps<BalanceParamList, 'SendScreen'>
+
+export function SendScreen ({ route }: Props): JSX.Element {
+  const [token] = useState(route.params.token)
+  const [isQRScanning, setIsQRScanning] = useState(false)
+  const { control, setValue, formState: { isValid }, getValues, trigger } = useForm({ mode: 'onChange' })
+
+  const onSubmit = (): void => {
+    console.log(isValid)
```
Comment (ivan-zynesis) [inline screens/AppNavigator/screens/BalancesScreen/SendScreen/SendScreen.tsx:41]: Will come back later after this part updated

File: screens/AppNavigator/screens/BalancesScreen/SendScreen/SendScreen.tsx (hunk 1 above)
Comment (thedoublejay) [reply screens/AppNavigator/screens/BalancesScreen/SendScreen/SendScreen.tsx:61]: @ivan-zynesis `useForm` API is from here - https://www.react-hook-form.com/. Basically, it's a hook to do form validations, like required, max amount and number pattern.

File: screens/AppNavigator/screens/BalancesScreen/SendScreen/SendScreen.tsx (hunk 2 above)
Comment (thedoublejay) [reply screens/AppNavigator/screens/BalancesScreen/SendScreen/SendScreen.tsx:73]: thanks! will check

**Comment by fuxingloh** [comment]
screenshot

DFI Token cannot be send, only DFI UTXO can be send

<details><summary>Browser metadata</summary>





```
Path:      /app/Balances
Browser:   Chrome 91.0.4472.114 on Mac OS 10.15.7
Viewport:  1680 x 971 @2x
Language:  en-GB
Cookies:   Enabled
```

[View on BrowserStack](https://live.browserstack.com/dashboard#url=https%3A%2F%2Fdeploy-preview-202--defi-wallet.netlify.app%2Fapp%2FBalances&os=OS+X&os_version=Catalina&browser=Chrome&browser_version=91.0&resolution=1680x971&speed=1&start=true)



</details>

[Open Deploy Preview](https://deploy-preview-202--defi-wallet.netlify.app/app/Balances) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=60d93ae11ff181000748a778&commentID=60dac83be98390680e9b919f&resolution=resolved)

File: components/index.tsx [hunk 4]
```diff
@@ -35,3 +45,17 @@ export function NumberText (props: { value: string, decimal?: number, style?: St
...
+      disabled={props.disabled}
+      onPress={props.onPress}
+      style={[tailwind('m-4 mt-8 p-3 rounded flex-row justify-center'), props.disabled === true ? DisabledColorStyle.button : PrimaryColorStyle.button]}
+    >
+      {
+        props.children
+      }
+    </TouchableOpacity>
+  )
+}
```
Comment (fuxingloh) [inline components/index.tsx:61]: should move this to an individual file

we need to start maintaining them as` components/*.tsx`

File: constants/Theme.ts [hunk 5]
```diff
@@ -15,6 +15,17 @@ export const PrimaryColorStyle = StyleSheet.create({
...
+  button: {
+    backgroundColor: PrimaryColor,
+    color: '#ffffff'
+  }
+})
+
+export const DisabledColorStyle = StyleSheet.create({
+  button: {
+    backgroundColor: 'rgba(0,0,0,0.2)',
+    color: 'rgba(255,255,255,0.06)'
```
Comment (fuxingloh) [inline constants/Theme.ts:28]: move this within the component as it should be scoped within that scope only

File: constants/Theme.ts [hunk 6]
```diff
@@ -15,6 +15,17 @@ export const PrimaryColorStyle = StyleSheet.create({
   },
   border: {
     borderColor: PrimaryColor
+  },
+  button: {
+    backgroundColor: PrimaryColor,
+    color: '#ffffff'
+  }
```
Comment (fuxingloh) [inline constants/Theme.ts:22]: don't use feature style, should only declare utility style

Comment (fuxingloh) [reply constants/Theme.ts:22]: aka more into component level

File: screens/AppNavigator/screens/BalancesScreen/SendScreen/SendScreen.tsx [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+import { getTokenIcon } from '../../../../../components/icons/tokens/_index'
+import { PrimaryColor, PrimaryColorStyle } from '../../../../../constants/Theme'
+import { translate } from '../../../../../translations'
+import { BalanceParamList } from '../BalancesNavigator'
+
+interface QRScannerProps {
+  onScan: (code: string) => void
+  isQRScanning: boolean
+  onCloseModal: () => void
+}
```
Comment (fuxingloh) [inline screens/AppNavigator/screens/BalancesScreen/SendScreen/SendScreen.tsx:20]: Move all QR features into another PR, don't prioritize this yet. We will track that individually.

**Review by fuxingloh — CHANGES_REQUESTED** [review]
> initial review

- move QR scanner into another Pr
- DFI UTXO can be send
- DFI Token cannot be send

File: screens/AppNavigator/screens/BalancesScreen/SendScreen/SendScreen.tsx (hunk 7 above)
Comment (thedoublejay) [reply screens/AppNavigator/screens/BalancesScreen/SendScreen/SendScreen.tsx:20]: Ok, do you want me to remove them here? It's already working well so I'm not sure what's the point of moving it out

File: constants/Theme.ts (hunk 6 above)
Comment (thedoublejay) [reply constants/Theme.ts:22]: ok copy

**Comment by thedoublejay** [comment]
@fuxingloh I have made some changes:
1. Disabled the click for DFI token
2. Removed usage of `NumberText` 
3. Moved PrimaryButton to dedicated component

I have left a comment for the QR scanner part. For me, I think we can leave it since it's already working well and it's part of the implementation. Let me know your thoughts about it

**Comment by fuxingloh** [comment]
> @fuxingloh I have made some changes:
> 
> 1. Disabled the click for DFI token
> 2. Removed usage of `NumberText`
> 3. Moved PrimaryButton to dedicated component
> 
> I have left a comment for the QR scanner part. For me, I think we can leave it since it's already working well and it's part of the implementation. Let me know your thoughts about it

Best to move it out, it's not maintainable for iterative reasons. Non-maintainable being: QR scanner has Modal component which is a design anti-pattern as it's easy to get wrong (TLDR don't use Modal in mobile). As this should be prioritized for merge, removing it for another PR will expedite the process.

**Comment by thedoublejay** [comment]
@fuxingloh ok I have removed the QR code changes

**Review by fuxingloh — APPROVED** [review]
